paf-mvp-frontend/tests/test-cases/queue.test.ts

```diff
@@ -0,0 +1,88 @@
+import { TestWatcher } from 'jest';
+import { isExportDeclaration } from 'typescript';
+import { Command, CommandQueue, processCommands } from '../../src/utils/queue';
+
+class CommandMocker {
+  public called = false;
+
+  func(): Command {
+    return () => {
+      this.called = true;
+    };
+  }
+}
+
+describe('queue', () => {
+  let cmd1: CommandMocker;
+  let cmd2: CommandMocker;
+  let cmd3: CommandMocker;
+  const lib = {
```

Comment:
You are testing the `queue` attribute, putting it inside a `lib` object only makes it more verbose.
OPT: use `const queue` instead and remove all the `lib.` in tests.

Reply:
I don't see how this is related to nested calls.
See my previous comment: I think what you need is just a `while` loop.

Reply:
Fixed.